src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,173 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTest {
+    private StorageDao storageDao;
+    private RegistrationService registrationService;
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }
```

Comment:
Consider if `@BeforeEach` is the best choice for initializing `RegistrationServiceImpl`. If the setup is expensive and does not change across tests, `@BeforeAll` with static initialization might be more efficient.